### ISSUE TITLE
SubtleCrypto.deriveBits fails with P-384 keys

### ISSUE THREAD

**Issue by diachedelic** [denoland/deno#14765]
The following script fails with an "Unsupported namedCurve" error. It works fine if the P-384 is changed to P-256.

```
function generate_keypair() {
    return crypto.subtle.generateKey(
        {name: "ECDH", namedCurve: "P-384"},
        true,
        ["deriveBits"]
    );
}

Promise.all([
    generate_keypair(),
    generate_keypair()
]).then(function ([alice, bob]) {
    return crypto.subtle.deriveBits(
        {name: "ECDH", public: bob.publicKey},
        alice.privateKey,
        256
    );
}).then(
    console.log
);
```

It appears that the upstream fix mentioned in [this TODO comment](https://github.com/denoland/deno/blob/7eee521199e9735ab7c347d99e9d90ba3046be1a/ext/crypto/lib.rs#L561) has just landed, meaning that it should be possible to make SubtleCrypto.deriveBits work with P-384 keys.

**Comment by diachedelic** [comment]
Version 0.11.0 of the __p384__ crate was released a few days ago, and it seems to contain the functionality we require. However, in bumping __p384__ from v0.8.0 to v0.11.0 I got some Cargo version conflicts. To resolve these conflicts, I upgraded some other RustCrypto crates:

- __elliptic-curve__ from 0.10.6 to 0.11.0 (0.12.0 is the latest)
- __p256__ from 0.9.0 to 0.11.0
- __rsa__ from 0.5.0 to 0.6.1
- __spki__ from 0.4.1 to 0.6.0

Unfortunately, this broke the build (https://gist.github.com/diachedelic/a760e057d032dcc53765dda73adf863f). At this point, I realised I was out of my depth.

**Comment by diachedelic** [comment]
I also noticed [this comment](https://github.com/denoland/deno/blob/94068b7109af64eba2cb9fd299e62705a5358069/ext/crypto/ec_key.rs#L57)

> Shamelessly copied from pkcs8 crate and modified so as to not require Arithmetic trait currently missing from p384

 which looks like it wants to be addressed alongside this issue.

**Comment by littledivy** [comment]
> Version 0.11.0 of the p384 crate was released a few days ago, and it seems to contain the functionality we require. 

There is a PR for upgrading all the webcrypto deps: https://github.com/denoland/deno/pull/14452 This should be unblocked after that.

**Comment by diachedelic** [comment]
Legend!

**Comment by diachedelic** [comment]
Duplicate of #13451.
